src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.cs

```diff
@@ -14,6 +18,70 @@ internal static void VerifyValidPath(string path, string argName)
             }
         }
 
+        internal static bool Copy(string sourcePath, string destinationPath, bool recursive,
+            bool skipExistingFiles, CancellationToken cancellationToken)
+        {
+            SearchOption searchOption = recursive ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly;
+            IEnumerable<string> directoryEnumeration;
+            bool success = true;
+
+            try
+            {
+                EnumerationOptions enumerationOptions = EnumerationOptions.Compatible;
+
+                if (recursive)
+                {
+                    enumerationOptions = EnumerationOptions.CompatibleRecursive;
+                }
+
+                directoryEnumeration = Directory.EnumerateDirectories(sourcePath, "*", enumerationOptions);
+            }
+            catch (IOException)
+            {
+                // Return false if Enumeration is not possible
+                return false;
+            }
+
+            foreach (string enumeratedDirectory in directoryEnumeration)
+            {
+                string newDirectoryPath = enumeratedDirectory.Replace(sourcePath, destinationPath);
+                Directory.CreateDirectory(newDirectoryPath);
+
+                cancellationToken.ThrowIfCancellationRequested();
+            }
+
+            foreach (string enumeratedFile in Directory.GetFiles(sourcePath, "*.*", searchOption))
+            {
+                if (skipExistingFiles && File.Exists(enumeratedFile))
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    continue;
+                }
+
+                try
+                {
+                    CopyFile(enumeratedFile, enumeratedFile.Replace(sourcePath, destinationPath), true);
+                }
+                catch (IOException ex)
```

Comment:
Same concern as previous catch, this should be more than IOException. e.g: UnauthorizedAccessException is the exception you would get for ERROR_ACCESS_DENIED.
https://github.com/dotnet/runtime/blob/440dfe4a7beecd7755767aa247f47af00b119383/src/libraries/Common/src/System/IO/Win32Marshal.cs#L39-L41